File: frontend/demo/component/avatar/avatar-image.ts
import 'Frontend/demo/init'; // hidden-source-line
import { html, LitElement } from 'lit';
import { customElement, state } from 'lit/decorators.js';
import '@vaadin/avatar';
import '@vaadin/horizontal-layout';
import { applyTheme } from 'Frontend/generated/theme';
import { getPeople } from 'Frontend/demo/domain/DataService';
import type Person from 'Frontend/generated/com/vaadin/demo/domain/Person';
import companyLogo from '../../../../src/main/resources/images/company-logo.png';

@customElement('avatar-image')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  @state()
  private person: Person | undefined;

  protected override async firstUpdated() {
    const { people } = await getPeople({ count: 1 });
    this.person = people[0];
  }

  protected override render() {
    return html`
      <vaadin-horizontal-layout theme="spacing">
        <!-- tag::snippet[] -->
        <vaadin-avatar
          .img="${this.person?.pictureUrl}"
          .name="${`${this.person?.firstName} ${this.person?.lastName}`}"
        ></vaadin-avatar>

        <vaadin-avatar .img="${companyLogo}" name="Company Inc."></vaadin-avatar>
        <!-- end::snippet[] -->
      </vaadin-horizontal-layout>
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/avatar/AvatarImage.java
package com.vaadin.demo.component.avatar;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.demo.domain.DataService;
import com.vaadin.demo.domain.Person;
import com.vaadin.flow.component.avatar.Avatar;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.router.Route;
import com.vaadin.flow.server.streams.DownloadHandler;

@Route("avatar-image")
public class AvatarImage extends HorizontalLayout {

    private Person person = DataService.getPeople(1).get(0);

    public AvatarImage() {
        String name = person.getFirstName() + " " + person.getLastName();
        String pictureUrl = person.getPictureUrl();

        // tag::snippet[]
        Avatar user = new Avatar(name);
        user.setImage(pictureUrl);

        Avatar company = new Avatar("Company Inc.");
        DownloadHandler imageHandler = DownloadHandler.forClassResource(
                getClass(), "/images/company-logo.png", "company-logo.png");
        company.setImageHandler(imageHandler);
        // end::snippet[]

        add(user, company);
    }

    public static class Exporter extends DemoExporter<AvatarImage> { // hidden-source-line
    } // hidden-source-line
}
